File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>

using namespace std;

int N, M;
vector<int> answer;
int graph[101][101];
bool visited[101]{};

void dfs(int v) {
	visited[v] = true;
	answer.emplace_back(v);
	// cout << v << '\n';
	for(int i = 1; i <= N; i++) {
		if(graph[v][i] == 1 && visited[i] == false) {
			dfs(i);
		}
	}
}

int main() {
	ios_base::sync_with_stdio(0);
	cin.tie(NULL);

	cin >> N >> M;

	for(int i = 0; i < M; i++) {
		int a, b;
		cin >> a >> b;
		graph[a][b] = 1;
		graph[b][a] = 1;
	}

	dfs(1);

	cout << answer.size() - 1 << '\n';
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

def dfs (graph, v, visited):
    visited[v] = 1
    for i in graph[v]:
        if visited[i] == 0:
            dfs(graph, i, visited)

#sys.stdin = open("input.txt")

n = int(sys.stdin.readline().rstrip())
e = int(sys.stdin.readline().rstrip())

graph = [[] for _ in range(n+1)]
visited = [0] * (n+1)

for i in range(e):
    start, end = map(int, sys.stdin.readline().split())
    graph[start].append(end)
    graph[end].append(start)

for i in range(1, n+1):
    graph[i].sort()

dfs(graph, 1, visited)

print(sum(visited)-1)


